secp256k1.nim

```diff
@@ -517,3 +517,17 @@ proc default*(T: type SkSignature): T {.error: "loophole".}
 proc default*(T: type SkRecoverableSignature): T {.error: "loophole".}
 proc default*(T: type SkEcdhSecret): T {.error: "loophole".}
 proc default*(T: type SkEcdhRawSecret): T {.error: "loophole".}
+
+func tweakAdd*(secretKey: var SkSecretKey, tweak: openArray[byte]): SkResult[void] =
+  {.noSideEffect.}: # secp256k1_context_no_precomp is actually const, see above
+    let res = secp256k1_ec_privkey_tweak_add(secp256k1_context_no_precomp, secretKey.data.ptr0, tweak.ptr0)
+    if res != 1:
```

Comment:
```suggestion
    if res != 1:
      err("...")
    else:
      ok()
```

`return` is redundant, and using it prevents disables some analysis the compiler does to detect branches missing `return` - we've had numerous security issues because of this - if it's not clearly making the code better, we avoid it (see rest of file)